tests/sentry/tasks/test_post_process.py

```diff
@@ -994,9 +897,159 @@ def test_ensure_when_assignees_and_owners_are_cached_does_not_cause_unbound_erro
+        ):
+            event = self.create_event(
+                data={"message": "testing", "fingerprint": ["group-1"]}, project_id=self.project.id
+            )
+            event_2 = self.create_event(
+                data={"message": "testing", "fingerprint": ["group-1"]}, project_id=self.project.id
+            )
+            group = event.group
+            group.update(times_seen=50)
+            snooze = GroupSnooze.objects.create(group=group, count=100, state={"times_seen": 0})
+
+            cache_key = write_event_to_cache(event)
+            post_process_group(
+                is_new=False,
+                is_regression=False,
+                is_new_group_environment=True,
+                cache_key=cache_key,
+                group_id=event.group_id,
+            )
+            assert GroupSnooze.objects.filter(id=snooze.id).exists()
+            cache_key = write_event_to_cache(event_2)
+            self.feature
+
+            buffer.incr(Group, {"times_seen": 60}, filters={"pk": event.group.id})
+            post_process_group(
+                is_new=False,
+                is_regression=False,
+                is_new_group_environment=True,
+                cache_key=cache_key,
+                group_id=event.group_id,
+            )
+            assert not GroupSnooze.objects.filter(id=snooze.id).exists()
+
+    @patch("sentry.rules.processor.RuleProcessor")
+    def test_maintains_valid_snooze(self, mock_processor):
+        event = self.create_event(data={}, project_id=self.project.id)
+        cache_key = write_event_to_cache(event)
+        group = event.group
+        snooze = GroupSnooze.objects.create(group=group, until=timezone.now() + timedelta(hours=1))
+
+        post_process_group(
+            is_new=True,
+            is_regression=False,
+            is_new_group_environment=True,
+            cache_key=cache_key,
+            group_id=event.group_id,
+        )
+
+        mock_processor.assert_called_with(EventMatcher(event), True, False, True, False)
+
+        assert GroupSnooze.objects.filter(id=snooze.id).exists()
+
+
 @region_silo_test
-class PostProcessGroupPerformanceTest(TestCase, SnubaTestCase, PerfIssueTransactionTestMixin):
-    @with_feature("organizations:performance-issues-post-process-group")
+class PostProcessGroupErrorTest(
+    TestCase,
+    AssignmentTestMixin,
+    CorePostProcessGroupTestMixin,
+    InboxTestMixin,
+    ResourceChangeBoundsTestMixin,
+    RuleProcessorTestMixin,
+    ServiceHooksTestMixin,
+    SnoozeTestMixin,
+):
+    def create_event(self, data, project_id):
+        return self.store_event(data=data, project_id=project_id)
+
+
+@region_silo_test
+@apply_feature_flag_on_cls("organizations:performance-issues-post-process-group")
+class PostProcessGroupPerformanceTest(
+    TestCase,
+    SnubaTestCase,
+    PerfIssueTransactionTestMixin,
+    CorePostProcessGroupTestMixin,
+    InboxTestMixin,
+    RuleProcessorTestMixin,
+    SnoozeTestMixin,
+):
```

Comment:
Should we extract out calls to `post_process_group` into a separate file and have it optionally overrideable? The main difference between processing errors vs transactions is the call args to `post_process_group`, i.e. `group_states`.

Reply:
That's a good idea, I'll make that change

Reply:
Oh damn... yeah, this isn't really written correctly in general, since I'm injecting the `GroupEvent` into the cache rather than the `Event`, and we're not relying on the passed `group_states` at all. Going to rework that part completely